easybuild/tools/filetools.py

```diff
@@ -303,6 +364,47 @@ def download_file(filename, url, path):
         return None
 
 
+def make_tarfile(output_filename, source_dir):
+    """Make a tarfile with the output_filename of the source_dir"""
+    _log.debug('making new tarfile at %s from %s', output_filename, source_dir)
+    with tarfile.open(output_filename, "w:gz") as tar:
+            tar.add(source_dir, arcname=os.path.basename(source_dir))
+    _log.debug('done making tarfile at %s', output_filename)
+
+
+def download_svn(revision, url, path, timeout):
+    """
+    Download a svn repository and create a tarball out of it in the given path
+    timeout is ignored here
+    """
+    # import here to avoid circular import
+    from easybuild.tools.repository.svnrepo import SvnRepository
+    # transform url into something SvnRepository understands
+    if url.startswith('svn+'):
+        url = url[4:]
+    # revision could already be a part of the url, we don't want that
+    if url.endswith(str(revision)):
+        url = url[:-len(revision)]
+    # same for path
+    if path.endswith(str(revision)):
+        path = path[:-len(revision)]
+    path = os.path.join(path, 'repo')
+    # svn revisions are digit's so only get these, some packages have version='r1234'
+    int_revision = ''.join([x for x in revision if x.isdigit()])
+
+    try:
+        svnrepo = SvnRepository(url)
+    except AttributeError, err:
+        _log.debug('error checkout out svn repo %s', err)
+        raise err
+    svnrepo.export(path, int_revision)
+    # make a tarfile in the directory next to the repo
+    make_tarfile(os.path.join(path, '..' ,revision), path)
+    # path will always have '/repo' in it, so the rm should be file
+    shutil.rmtree(os.path.join(path))
+    return os.path.join(path, '..', revision)
```

Comment:
use a variable for `os.path.join(pat, '..', revision)`, you're using it in two places